packages/storage-plus/src/snapshot.rs

```diff
@@ -297,24 +300,24 @@ mod tests {
     // Values at beginning of 3 -> A = 5, B = 7
     // Values at beginning of 5 -> A = 8, C = 13
     fn init_data(map: &TestMap, storage: &mut dyn Storage) {
-        map.save(storage, b"A", &5, 1).unwrap();
-        map.save(storage, b"B", &7, 2).unwrap();
+        map.save(storage, &b"A".as_ref(), &5, 1).unwrap();
```

Comment:
This is introducing double references (`&&`). You could be defining the key as a value (`[u8]`) to avoid that; and all these would look nicer.

Reply:
If you define the `TestMap` using `[u8]` both references can be removed, by the way. One will be removed explicitly by you, the other will be "added" (deducted) by the compiler, and therefore does not need to be specified.
```suggestion
        map.save(storage, b"A", &5, 1).unwrap();
```

Reply:
`[u8]` (`u8` slice of unknown size) can be used if you have `K: ?Sized` (i.e. remove the `Sized` constraint that exists by default), see https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.get.

If you want to deserialize into this unsized `K`, the `parse_key` needs to put it on the heap and return something like `Box<K>` instead of `K`.

Reply:
That said, syntax is a bit uglier, but I think we can live with that, if we agree this solution is better in the end, in that it forces users to use references.